src/components/Tooltip/Tooltip.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+    <div
+        ref="tooltip"
+        class="d-inline-block"
+    >
+        <slot v-if="!!$slots.default" />
+    </div>
+</template>
+
+<script lang="ts">
+import {PropType, computed, ref} from 'vue';
+import {tooltipFallbackPlacementProps, tooltipPlacementProps} from '@/composables/useTooltipPlacement';
+import {triggerProps} from '@/composables/useTrigger';
+import useBootstrapEmits from '@/composables/useBootstrapEmits';
+import useBootstrapInstance from '@/composables/useBootstrapInstance';
+
+const tooltipEvents = ['show', 'shown', 'hide', 'hidden', 'inserted'] as const;
+
+type TooltipEvent = typeof tooltipEvents[number];
+</script>
+
+<script lang="ts" setup>
+const props = defineProps({
+    boundary: {
+        type: [String, HTMLElement],
+        default: 'clippingParents',
+    },
+    container: {
+        type: [String, HTMLElement, Boolean],
+        default: false,
+    },
+    customClass: {
+        type: [String, Array, Object],
+        default: '',
+    },
+    delay: {
+        type: [Number, Object],
+        default: 0,
+    },
+    html: {
+        type: Boolean,
+        default: false,
+    },
+    offset: {
+        type: [String, Array] as PropType<string | number[]>,
+        default: () => [0, 0],
+    },
+    popperConfig: {
+        type: [Object, Function],
+        default: () => ({}),
+    },
+    title: {
+        type: String,
+        default: '',
+    },
+    ...tooltipFallbackPlacementProps,
+    ...tooltipPlacementProps,
+    ...triggerProps,
+});
+
+const tooltip = ref<HTMLElement>();
+
+const emit = defineEmits<{(event: TooltipEvent): void}>();
+
+const tooltipClass = computed(() => {
+    if (typeof props.customClass === 'string') {
+        return props.customClass;
+    }
+
+    if (Array.isArray(props.customClass)) {
+        return props.customClass.join(' ');
+    }
+
+    return Object.keys(props.customClass)
+        .filter((key: string) => !!props.customClass[key])
+        .join(' ');
+});
```

Comment:
Why do you need to implement this logic instead of using Vue's feature?

Reply:
Because the `customClass` property of the [bootstrap Tooltip options](https://getbootstrap.com/docs/5.1/components/tooltips/#options) only supports a String or a function. This doesn't work like `:class="customClass"` would

![image](https://user-images.githubusercontent.com/18048711/177151840-02821e92-69de-4c5e-baab-3c43cf74da8a.png)